ffi/diplomat/src/properties_maps.rs

```diff
@@ -6,41 +6,151 @@
 pub mod ffi {
     use crate::provider::ffi::ICU4XDataProvider;
     use alloc::boxed::Box;
-    use icu_properties::{maps, Script};
+    use icu_collections::codepointtrie::TrieValue;
+    use icu_properties::{maps, PropertiesError};
 
     use crate::errors::ffi::ICU4XError;
     use diplomat_runtime::DiplomatResult;
 
+    use icu_provider::prelude::BufferProvider;
+    use icu_provider::serde::{AsDeserializingBufferProvider, DeserializingBufferProvider};
+
     #[diplomat::opaque]
-    /// An ICU4X Unicode Set Property object, capable of querying whether a code point is contained in a set based on a Unicode property. For properties whose values fit into 16 bits.
+    /// An ICU4X Unicode Set Property object, capable of querying whether a code point is contained in a set based on a Unicode property.
+    ///
+    /// For properties whose values fit into 8 bits.
     #[diplomat::rust_link(icu::properties, Mod)]
-    pub struct ICU4XCodePointMapData16(maps::CodePointMapData<Script>);
+    #[diplomat::rust_link(icu::properties::maps::CodePointMapData, Struct)]
+    #[diplomat::rust_link(icu::properties::maps::CodePointMapDataBorrowed, Struct)]
```

Comment:
Praise: I like adding multiple rust links. I think we should do this in other places as well, in order to get the coverage up.

Reply:
Also, over FFI, I think it's okay to have a single type here. The borrowed type is a Rust-side optimization. I would be surprised if it's big enough to show up in cross-FFI benchmarks.